bot/exts/backend/error_handler.py

```diff
@@ -0,0 +1,159 @@
+import math
+import random
+from typing import Iterable, Union
+
+import discord
+from bot.constants import Channels, Colours, ERROR_REPLIES, NEGATIVE_REPLIES
+from discord import Embed, Message
+from discord.ext import commands
+from loguru import logger
+
+
+class CommandErrorHandler(commands.Cog):
+    """Handles errors emitted from commands."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+    @staticmethod
+    def revert_cooldown_counter(command: commands.Command, message: Message) -> None:
+        """Undoes the last cooldown counter for user-error cases."""
+        if command._buckets.valid:
+            bucket = command._buckets.get_bucket(message)
+            bucket._tokens = min(bucket.rate, bucket._tokens + 1)
+            logger.debug(
+                "Cooldown counter reverted as the command was not used correctly."
+            )
+
+    @staticmethod
+    def error_embed(message: str, title: Union[Iterable, str] = ERROR_REPLIES) -> Embed:
+        """Build a basic embed with red colour and either a random error title or a title provided."""
+        embed = Embed(colour=Colours.soft_red)
+        if isinstance(title, str):
+            embed.title = title
+        else:
+            embed.title = random.choice(title)
+        embed.description = message
+        return embed
+
+    @commands.Cog.listener()
+    async def on_command_error(
+        self, ctx: commands.Context, error: commands.CommandError
+    ) -> None:
+        """Activates when a command opens an error."""
+        if getattr(error, "handled", False):
+            logger.debug(
+                f"Command {ctx.command} had its error already handled locally; ignoring."
+            )
+            return
+
+        error = getattr(error, "original", error)
+
+        # Skip logging CommandNotFound Error
+        if isinstance(error, commands.CommandNotFound):
+            return
+
+        logger.debug(
+            f"Error Encountered: {type(error).__name__} - {str(error)}, "
+            f"Command: {ctx.command}, "
+            f"Author: {ctx.author}, "
+            f"Channel: {ctx.channel}"
+        )
+
+        if isinstance(error, commands.UserInputError):
```

Comment:
```suggestion
       if isinstance(error, commands.CommandNotFound):
            return  # exiting early to avoid logging
        elif isinstance(error, commands.UserInputError):
```
After taking my suggestions into consideration, `CommandNotFound` can be part of the `if-elif-else` block again.